source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -504,13 +504,6 @@ void GlslShaderGenerator::emitPixelStage(const ShaderGraph& graph, GenContext& c
     emitInclude("pbrlib/" + GlslShaderGenerator::LANGUAGE + "/lib/mx_math.glsl", context, stage);
     emitLineBreak(stage);
 
-    // Emit texture sampling code
-    if (graph.hasClassification(ShaderNode::Classification::CONVOLUTION2D))
-    {
-        emitInclude("stdlib/" + GlslShaderGenerator::LANGUAGE + "/lib/mx_sampling.glsl", context, stage);
```

Comment:
Move this include from shader generators into each individual nodes that needs it, so we avoid needing to track if a node is convolution or not.